code/PrepareDataForSubmarineCable.py

```diff
@@ -0,0 +1,173 @@
+    linestrings = multilinestring.replace(
+        'MULTILINESTRING ', '').strip('()').split('), (')
+    # linestrings = linestrings.replace('"', '')
+    list_of_linestring_lists = []  # This will be the final list of lists to return
+
+    # Iterate over each linestring
+    for linestring in linestrings:
+        points = linestring.split(', ')  # Split the linestring into points
+        list_of_points = []  # This will hold the tuples for the current linestring
+
+        # Iterate over each point in the linestring
+        for point in points:
+            # Split the point into its x and y components and make sure to strip any potential white space
+            # which might lead to a ValueError if trying to convert an empty string to float
+            xy = point.strip().split(' ')
+            if xy:  # Proceed only if the split is non-empty
+                # Convert the x and y string components to float
+                x, y = map(float, xy)
+                # Append the tuple to the list of points
+                list_of_points.append((x, y))
+
+        # After all points of a linestring are processed, append the list of points to the list of linestring lists
+        list_of_linestring_lists.append(list_of_points)
+
+    return list_of_linestring_lists
+
+
+def convert_land_multilinestring_to_list(multilinestring):
+    # Remove the 'MULTILINESTRING ' prefix and split the string into individual linestrings
+    linestrings = multilinestring.replace(
+        'LINESTRING', '').strip('()').split('), (')
+    # linestrings = linestrings.replace('"', '')
+    list_of_linestring_lists = []  # This will be the final list of lists to return
+
+    # Iterate over each linestring
+    for linestring in linestrings:
+        points = linestring.split(', ')  # Split the linestring into points
+        list_of_points = []  # This will hold the tuples for the current linestring
+
+        # Iterate over each point in the linestring
+        for point in points:
+            # Split the point into its x and y components and make sure to strip any potential white space
+            # which might lead to a ValueError if trying to convert an empty string to float
+            xy = point.strip().split(' ')
+            if xy:  # Proceed only if the split is non-empty
+                # Convert the x and y string components to float
+                x, y = map(float, xy)
+                # Append the tuple to the list of points
+                list_of_points.append((x, y))
+
+        # After all points of a linestring are processed, append the list of points to the list of linestring lists
+        list_of_linestring_lists.append(list_of_points)
+
+    return list_of_linestring_lists
+
+# Correcting the structure of the provided MULTILINESTRING data to properly format it into KML
+# Removing the extra for loop which was causing the unpacking error
+
+
+def create_file(whole_data):
```

Comment:
It might make sense to have a separate file for creating KML docs

Reply:
maybe we can merge them to one file later.